CONTRIBUTING.md

```diff
@@ -301,37 +243,83 @@ This is optional but encouraged.
-- Gateway Protocol Implementation ([gateway-protocol-impl](/gateway-protocol-impl))
-- BPMN Model API ([bpmn-model](/bpmn-model))
+For contributions to Camunda, building quickly is typically sufficient.
+However, users of Camunda are recommended to build the full distribution.
 
-If you would like to contribute something, or simply want to hack on the code
-this document should help you get started.
+To fully build the Zeebe distribution, run the command: `mvn clean install -DskipTests` in the root folder.
+This is slightly slower than building quickly but ensures the distribution is assembled completely.
+The resulting Zeebe distribution can be found in the folder `dist/target`, i.e.
 
-## Code of Conduct
+```
+dist/target/camunda-zeebe-X.Y.Z-SNAPSHOT.tar.gz
+dist/target/camunda-zeebe-X.Y.Z-SNAPSHOT.zip
+```
 
-This project adheres to the [Camunda Code of Conduct](https://camunda.com/events/code-conduct/).
-By participating, you are expected to uphold this code. Please [report](https://camunda.com/events/code-conduct/reporting-violations/)
-unacceptable behavior as soon as possible.
+This distribution can be containerized with Docker (i.e. build a Docker image) by running:
+
+```
+docker build \
+  --tag camunda/zeebe:local \
+  --build-arg DISTBALL='dist/target/camunda-zeebe*.tar.gz' \
+  --target app \
+  .
+```
+
+This is a small overview of the contents of the different modules:
+- `util` contains custom implementations of building blocks like an actor scheduler, buffer allocations, and metrics. Its parts are used in most of the other modules
+- `protocol` contains the SBE definition of the main message protocol
+- `bpmn-model` is a Java API for BPMN process definitions used for parsing etc.
+- `msgpack-*` is a custom msgpack implementation with extensions to evaluate json-path expressions on msgpack objects
+- `dispatcher` is a custom implementation of message passing between threads
+- `service-container` is a custom implementation to manage dependencies between different services
+- `logstreams` is an implementation of an append-only log backed by the filesystem
+- `transport` is our abstraction over network transports
+- `gateway` is the implementation of the gRPC gateway, using our SBE-based protocol to communicate with brokers
+- `gateway-protocol` contains the gRPC definitions for the Zeebe client-to-gateway protocol
+- `zb-db` is our RocksDB wrapper for state management
+- `engine`  is the implementation of the event stream processor
+- `broker` contains the Zeebe broker which is the server side of Zeebe
+- `client-java` contains the Java Zeebe client
+- `atomix` contains transport, membership, and consensus algorithms
+- `benchmark` contains utilities the team uses to run load tests
+- `exporters/elasticsearch-exporter` contains the official Elasticsearch exporter for Zeebe
+- `journal` contains the append-only log used by the consensus algorithm
+- `snapshots` abstracts how state snapshots (i.e. `zb-db`) are handled
```

Comment:
🔧 Some of these modules have been moved into the `zeebe` subfolder. I suggest we adjust this list of modules to explain the root-level folders, and move the Zeebe module specific ones into a README in the zeebe subfolder

Reply:
👍

Reply:
Moved the Zeebe-specific folders to a Zeebe README, and added what I could to this section. I'm not sure what some folders are or if they need to be explained.